app/models/rule_group_relationship.rb

```diff
@@ -13,4 +13,8 @@ class RuleGroupRelationship < ApplicationRecord
   scope :with_relationships, lambda { |rules_or_groups, relationship|
     where(relationship: relationship).includes(:left, :right).where(left: rules_or_groups)
   }
+
+  scope :with_distinct_relationships, lambda { |rules_or_groups, relationship|
+    where(left: rules_or_groups, relationship: relationship).select(:right_id, :right_type).distinct
```

Comment:
@skateman `.select(:right)` doesn't work so I did `.select(:right_id, :right_type)` but this is faster than using `uniq` 👍

Reply:
No need to name it `_distinct`, enough to have it as `with_relationships` as it's shorter.

Reply:
There already is a `with_relationships` scope :/ I was going to just add the `.distinct` part onto that one but we're already using `with_relationships` and we don't only want the distinct relationships so that's why I had to create a separate scope for this

Reply:
As I see it's similar, so you could just do `with_relationship(...).distinct` or something like that instead. This way you don't need the separate scope.

Reply:
Ah right, I'll just do that instead